### PR TITLE
cors: remove unnecessary code

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by wbpcode** [comment]
Found this when I review your PR cc @adisuissa make the envoy better a little everyday 🤣

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Review by adisuissa — COMMENTED** [review]
CI seems to strongly disagree.
/wait

**Comment by wbpcode** [comment]
I initially only want do a simple thing...

File: test/extensions/filters/http/cors/cors_filter_test.cc [hunk 1]
```diff
@@ -60,17 +62,17 @@ class CorsFilterTest : public testing::Test {
...
 
-  bool isCorsRequest() { return filter_.is_cors_request_; }
+  bool isCorsRequest() { return filter_->is_cors_request_; }
 
   NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks_;
   NiceMock<Http::MockStreamEncoderFilterCallbacks> encoder_callbacks_;
   Stats::TestUtil::TestStore stats_;
   CorsFilterConfigSharedPtr config_;
-  CorsFilter filter_;
+  std::shared_ptr<CorsFilter> filter_;
```
Comment (adisuissa) [inline test/extensions/filters/http/cors/cors_filter_test.cc:75]: Why is this change needed?
If possible, I suggest leaving this as `CorsFilter` type.

**Review by adisuissa — COMMENTED** [review]
Nice, thanks!
Minor comment about the test, but otherwise, LGTM!

File: test/extensions/filters/http/cors/cors_filter_test.cc (hunk 1 above)
Comment (wbpcode) [reply test/extensions/filters/http/cors/cors_filter_test.cc:75]: compiler complain that because now it inherit from passthroughfilter. Compiler will throw an error about multiple assignments of virtual base.

Comment (wbpcode) [reply test/extensions/filters/http/cors/cors_filter_test.cc:75]: here is the error https://github.com/envoyproxy/envoy/actions/runs/11956111680/job/33330127779

I actually dislike to inherit from multiple classes and virtual inheritance. But anyway, it's widely used by envoy and almost all stream filters inherit from passthroughfilter. So, I think it's fine for now, before we can get better abstraction/solution.

Comment (adisuissa) [reply test/extensions/filters/http/cors/cors_filter_test.cc:75]: I guess that the main issue is that the places where the CorsFilter is moved in the tests actually don't need to have the filter as a data member (just create a locally scoped filter `CorsFilter filter(config_);` instead of updating the `filter_`).
As this is just a test code, I don't fell strongly about using a pointer, I just think it makes more sense in this case to not use one.

**Review by adisuissa — APPROVED** [review]
Left a comment, but approving as this isn't something too important.

**Comment by adisuissa** [comment]
Ugh.. missed the fact that this wPR had auto-submit enabled.

File: test/extensions/filters/http/cors/cors_filter_test.cc (hunk 1 above)
Comment (wbpcode) [reply test/extensions/filters/http/cors/cors_filter_test.cc:75]: I also didn't feel strongly about that and only did a global replacement.
